integrated_channels/canvas/models.py

```diff
@@ -0,0 +1,135 @@
+from integrated_channels.integrated_channel.models import EnterpriseCustomerPluginConfiguration
+
+LOGGER = getLogger(__name__)
+
+
+# pylint: disable=feature-toggle-needs-doc
+@python_2_unicode_compatible
+class CanvasGlobalConfiguration(ConfigurationModel):
+    """
+    The global configuration for integrating with Canvas.
+
+    .. no_pii:
+    """
+
+    course_api_path = models.CharField(
+        max_length=255,
+        verbose_name="Course Metadata API Path",
+        help_text="The API path for making course metadata POST/DELETE requests to Canvas."
+    )
+
+    oauth_api_path = models.CharField(
+        max_length=255,
+        verbose_name="OAuth API Path",
+        default='login/oauth2/token',
+        help_text=(
+            "The API path for making OAuth-related POST requests to Canvas. "
+            "This will be used to gain the OAuth access token for other API calls."
+        )
+    )
+
+    class Meta:
+        app_label = 'canvas'
+
+    def __str__(self):
+        """
+        Return a human-readable string representation of the object.
+        """
+        return "<CanvasGlobalConfiguration with id {id}>".format(id=self.id)
+
+    def __repr__(self):
+        """
+        Return uniquely identifying string representation.
+        """
+        return self.__str__()
+
+
+@python_2_unicode_compatible
+class CanvasEnterpriseCustomerConfiguration(EnterpriseCustomerPluginConfiguration):
+    """
+    The Enterprise-specific configuration we need for integrating with Canvas.
+
+    Based on: https://canvas.instructure.com/doc/api/file.oauth.html#oauth2-flow-3
+
+    .. no_pii:
+    """
+
+    key = models.CharField(
+        max_length=255,
+        blank=True,
+        verbose_name="API Client ID",
+        help_text=(
+            "The API Client ID provided to edX by the enterprise customer to be used to make API "
+            "calls to Canvas on behalf of the customer."
+        )
+    )
+
+    secret = models.CharField(
+        max_length=255,
+        blank=True,
+        verbose_name="API Client Secret",
+        help_text=(
+            "The API Client Secret provided to edX by the enterprise customer to be used to make "
+            " API calls to Canvas on behalf of the customer."
+        )
+    )
+
+    canvas_company_id = models.CharField(
+        max_length=255,
+        blank=True,
+        verbose_name="Canvas Organization Code",
+        help_text="The organization code provided to the enterprise customer by Canvas."
+    )
+
+    canvas_base_url = models.CharField(
+        max_length=255,
+        blank=True,
+        verbose_name="Canvas Base URL",
+        help_text="The base URL used for API requests to Canvas, i.e. https://instructure.com."
+    )
+
+    provider_id = models.CharField(
+        max_length=100,
+        default='EDX',
+        verbose_name="Provider Code",
+        help_text="The provider code that Canvas gives to the content provider."
+    )
+
+    history = HistoricalRecords()
```

Comment:
I did not quite get why for Degreed this model has a historicalRecords added but not for the other models, I just blindly followed that example, but hopefully one of the reviewers can enlighten me here; if the extra complexity is not needed we should reduce it .. but if this is a need let's keep it

Reply:
yeah I just see that if we don't use historicalrecords it seems that we get a new row on each save() call, and current() will return the latest record. I figure when you work on the models in the next ticket, we should identify if we need any unique key fields or primary key fields in these models and update accordingly

My understanding is that: There will only be one record for a CanvasGlobalConfiguration (not per enterprise) assuming all canvas installations work the same way ..?

But the per enterprise config is going to be .. ahem.. per enterprise .. and I presume will undergo more frequent changes, hence it is wired up to historical records? I am making up my own reasoning here but that is one reason I can see for the difference

Reply:
in any case I doubt this blocks this PR since the purpose here was to setup something we can work with, nothing works right now here anyway :)